src/app/view/editor-main-view/data-views/trainrunsections.view.ts

```diff
@@ -733,10 +852,64 @@ export class TrainrunSectionsView {
   }
 
   static getTrainrunSectionValueToShow(
-    trainrunSection: TrainrunSection,
+    trainrunSectionOrViewObject: TrainrunSection | TrainrunSectionViewObject,
     textElement: TrainrunSectionText,
     editorView: EditorView,
   ) {
+    // Determine which section to use
+    let trainrunSection: TrainrunSection;
+
+    if (trainrunSectionOrViewObject instanceof TrainrunSection) {
+      // Simple case: we have a TrainrunSection directly
+      trainrunSection = trainrunSectionOrViewObject;
+    } else {
+      // ViewObject case: determine which section to use based on textElement
+      const viewObject = trainrunSectionOrViewObject;
+
+      switch (textElement) {
+        case TrainrunSectionText.SourceDeparture:
+        case TrainrunSectionText.SourceArrival:
+          trainrunSection = viewObject.trainrunSections[0];
+          break;
+
+        case TrainrunSectionText.TargetDeparture:
+        case TrainrunSectionText.TargetArrival:
+          trainrunSection = viewObject.trainrunSections.at(-1)!;
+          break;
+
+        case TrainrunSectionText.TrainrunSectionTravelTime:
+          // Special case for multiple sections: calculate total time including stop times at intermediate nodes
+          if (viewObject.trainrunSections.length > 1) {
+            const totalTime = viewObject.trainrunSections.reduce((sum, section, index) => {
```

Comment:
We could move this into `TrainrunSectionViewObject.getTravelTime()`, and call it from here. That way, we fix bugs in `getTravelTime()` and reduce code duplication.